src/xrpld/ledger/detail/View.cpp

```diff
@@ -1662,4 +1882,84 @@ deleteAMMTrustLine(
     return tesSUCCESS;
 }
 
+TER
+rippleCreditMPT(
+    ApplyView& view,
+    AccountID const& uSenderID,
+    AccountID const& uReceiverID,
+    STAmount const& saAmount,
+    beast::Journal j)
+{
+    auto const mptID = keylet::mptIssuance(saAmount.get<MPTIssue>().getMptID());
+    auto const issuer = saAmount.getIssuer();
+    if (!view.exists(mptID))
+        return tecOBJECT_NOT_FOUND;
+    if (uSenderID == issuer)
+    {
+        if (auto sle = view.peek(mptID))
+        {
+            sle->setFieldU64(
+                sfOutstandingAmount,
+                sle->getFieldU64(sfOutstandingAmount) + saAmount.mpt().value());
```

Comment:
IMHO, there are two advantages:
1. Less code.
2. Less repeated code. Specifically, this way only requires the field name (`sfOutstandingAmount`) to be provided once.

You're right, though, that you have to know that `amt` is a `ValueProxy`, which can do read _and write_ operations on a field. It might be more obvious and readable if we could specify the type as `STObject::ValueProxy<STInt64>` instead of `auto`, but it looks like the `Proxy` classes are specified as `private`! I don't know why that is. They could probably be made `public` without too much issue.

It might also be more obvious and readable if `ValueProxy` had that `+=` operator I was talking about earlier, because then you could just write it as
```
sle->at(sfOutstandingAmount) += saAmount.mpt().value();
```
And it would be very clear what was going on.

Reply:
I'm going to add += and -= to `ValueProxy` and simplify some of the MPT code in `View`.